worker/rhpv2.go

```diff
@@ -598,8 +598,10 @@ func (w *worker) fetchContractRoots(t *rhpv2.Transport, rev *rhpv2.ContractRevis
 		rev.Signatures[1].Signature = rootsResp.Signature[:]
 
 		// verify the proof
-		if !rhpv2.VerifySectorRangeProof(rootsResp.MerkleProof, rootsResp.SectorRoots, offset, offset+n, numsectors, rev.Revision.FileMerkleRoot) {
-			return nil, fmt.Errorf("could verify roots proof, host %v, version %v; %w", rev.HostKey(), settings.Version, ErrInvalidMerkleProof)
+		if uint64(len(rootsResp.SectorRoots)) != n {
+			return nil, fmt.Errorf("couldn't verify contract roots proof, host %v, version %v, err: number of roots does not match range %d != %d (num sectors: %d offset: %d)", rev.HostKey(), settings.Version, len(rootsResp.SectorRoots), n, numsectors, offset)
```

Comment:
Might be useful to also print the size of the revision we are using. That way we can check if the number of roots we receive matches the size of the revision and if not it would indicate a race.

Reply:
Good suggestion thanks!